Okta Provider cannot be built for C# (`System` property shadows another `System`)

<!--- Please keep this note for the community --->

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### cdktf & Language Versions
0.8
<!--- Please make note of the version you are using for the `hashicorp/terraform-cdk` and language bindings you are using.
If you are not running the latest version, please upgrade because your issue may have already been fixed. --->

### Affected Resource(s)
`HashiCorp/Cdktf/Providers/Okta/DataOktaAuthServerScopesScopes.cs`

<!--- Please list the affected resources. --->

### Debug Output

```
[jsii-pacmak] [WARN] Exception occurred, not cleaning up /var/folders/19/0g2813wd0g77r4dvy2_xtmfw0000gq/T/npm-packnAJOtx
[jsii-pacmak] [WARN] dotnet failed
Error: All attempts failed. Last error: Command (dotnet build --force --configuration Release) failed with status 1:
#STDERR>
#STDOUT> Microsoft (R) Build Engine version 16.9.0+57a23d249 for .NET
#STDOUT> Copyright (C) Microsoft Corporation. All rights reserved.
#STDOUT>
#STDOUT>   Determining projects to restore...
#STDOUT>   Restored /private/var/folders/19/0g2813wd0g77r4dvy2_xtmfw0000gq/T/npm-packnAJOtx/HashiCorp.Cdktf.Providers.Okta/HashiCorp.Cdktf.Providers.Okta.csproj (in 172 ms).
#STDOUT> /private/var/folders/19/0g2813wd0g77r4dvy2_xtmfw0000gq/T/npm-packnAJOtx/HashiCorp.Cdktf.Providers.Okta/HashiCorp/Cdktf/Providers/Okta/DataOktaAuthServerScopesScopes.cs(19,48): error CS0120: An object reference is required for the non-static field, method, or property 'DataOktaAuthServerScopesScopes.System' [/private/var/folders/19/0g2813wd0g77r4dvy2_xtmfw0000gq/T/npm-packnAJOtx/HashiCorp.Cdktf.Providers.Okta/HashiCorp.Cdktf.Providers.Okta.csproj]
#STDOUT> /private/var/folders/19/0g2813wd0g77r4dvy2_xtmfw0000gq/T/npm-packnAJOtx/HashiCorp.Cdktf.Providers.Okta/HashiCorp/Cdktf/Providers/Okta/DataOktaAuthServerScopesScopes.cs(26,48): error CS0120: An object reference is required for the non-static field, method, or property 'DataOktaAuthServerScopesScopes.System' [/private/var/folders/19/0g2813wd0g77r4dvy2_xtmfw0000gq/T/npm-packnAJOtx/HashiCorp.Cdktf.Providers.Okta/HashiCorp.Cdktf.Providers.Okta.csproj]
#STDOUT>
#STDOUT> Build FAILED.
#STDOUT>
#STDOUT> /private/var/folders/19/0g2813wd0g77r4dvy2_xtmfw0000gq/T/npm-packnAJOtx/HashiCorp.Cdktf.Providers.Okta/HashiCorp/Cdktf/Providers/Okta/DataOktaAuthServerScopesScopes.cs(19,48): error CS0120: An object reference is required for the non-static field, method, or property 'DataOktaAuthServerScopesScopes.System' [/private/var/folders/19/0g2813wd0g77r4dvy2_xtmfw0000gq/T/npm-packnAJOtx/HashiCorp.Cdktf.Providers.Okta/HashiCorp.Cdktf.Providers.Okta.csproj]
#STDOUT> /private/var/folders/19/0g2813wd0g77r4dvy2_xtmfw0000gq/T/npm-packnAJOtx/HashiCorp.Cdktf.Providers.Okta/HashiCorp/Cdktf/Providers/Okta/DataOktaAuthServerScopesScopes.cs(26,48): error CS0120: An object reference is required for the non-static field, method, or property 'DataOktaAuthServerScopesScopes.System' [/private/var/folders/19/0g2813wd0g77r4dvy2_xtmfw0000gq/T/npm-packnAJOtx/HashiCorp.Cdktf.Providers.Okta/HashiCorp.Cdktf.Providers.Okta.csproj]
#STDOUT>     0 Warning(s)
#STDOUT>     2 Error(s)
#STDOUT>
#STDOUT> Time Elapsed 00:00:01.63
#STDOUT>
    at retry (/Users/ansgar/projects/hashicorp/cdktf-provider-okta/node_modules/jsii-pacmak/lib/util.js:128:27)
    at async DotnetBuilder.buildModules (/Users/ansgar/projects/hashicorp/cdktf-provider-okta/node_modules/jsii-pacmak/lib/targets/dotnet.js:39:13)
    at async Promise.all (index 0)
    at async Object.pacmak (/Users/ansgar/projects/hashicorp/cdktf-provider-okta/node_modules/jsii-pacmak/lib/index.js:58:9)
```
<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.
--->

### Expected Behavior

should produce valid C# code.

<!--- What should have happened? --->

### Actual Behavior
Fails with `error CS0120: An object reference is required for the non-static field, method, or property 'DataOktaAuthServerScopesScopes.System'`
<!--- What actually happened? --->

### Steps to Reproduce
Clone prebuilt provider repo for Okta Terraform provider and checkout this PR: https://github.com/hashicorp/cdktf-provider-okta/pull/2
Run `yarn install && yarn build`

<!--- Please list the steps required to reproduce the issue. --->

### Important Factoids

This code (from `HashiCorp.Cdktf.Providers.Okta/HashiCorp/Cdktf/Providers/Okta/DataOktaAuthServerScopesScopes.cs`)
```csharp
/// <summary>Used by jsii to construct an instance of this class from a Javascript-owned object reference</summary>
/// <param name="reference">The Javascript-owned object reference</param>
[System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
protected DataOktaAuthServerScopesScopes(ByRefValue reference): base(reference)
{
}
```
fails because `System.ComponentModel.[...]` is shadowed by an attribute specified further down:
```csharp
[JsiiProperty(name: "system", typeJson: "{\"primitive\":\"any\"}")]
public virtual object System
{
    get => GetInstanceProperty<object>()!;
}
```
<!--- Are there anything atypical about your accounts that we should know? --->

### References
* https://github.com/hashicorp/cdktf-provider-okta/runs/4521130544?check_suite_focus=true

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

--- a/packages/@cdktf/provider-generator/lib/get/__tests__/generator/fixtures/incompatible-attribute-names.test.fixture.json
+++ b/packages/@cdktf/provider-generator/lib/get/__tests__/generator/fixtures/incompatible-attribute-names.test.fixture.json
@@ -17,6 +17,14 @@
               "equals": {
                 "type": "string",
                 "required": true
+              },
+              "node": {
+                "type": "string",
+                "required": true
+              },
+              "system": {
+                "type": "string",
+                "required": true
               }
             }
           },

--- a/packages/@cdktf/provider-generator/lib/get/generator/models/attribute-model.ts
+++ b/packages/@cdktf/provider-generator/lib/get/generator/models/attribute-model.ts
@@ -188,6 +188,8 @@ export class AttributeModel {
     if (this._name === "equals") return "equalTo";
     // `node` is already used by the Constructs base class
     if (this._name === "node") return "nodeAttribute";
+    // `System` shadows built-in types in CSharp (see #1420)
+    if (this._name === "system") return "systemAttribute";
     // `tfResourceType` is already used by resources to distinguish between different resource types
     if (this._name === "tfResourceType") return `${this._name}Attribute`;
     return this._name;

